lib/Sema/CSGen.cpp

```diff
@@ -3280,7 +3280,8 @@ namespace {
       if (auto keyPath = dyn_cast<KeyPathExpr>(expr)) {
         if (keyPath->isObjC()) {
           auto &cs = CG.getConstraintSystem();
-          (void)cs.getTypeChecker().checkObjCKeyPathExpr(cs.DC, keyPath);
+          if (!cs.getTypeChecker().checkObjCKeyPathExpr(cs.DC, keyPath))
```

Comment:
Do we emit a diagnostic when this returns false?

Reply:
Yes, at least in the cases we already hit this in during test runs.